### PR TITLE
BUG - False drops groupings

### LINKED ISSUE #58337: BUG: groupby with as_index=False drops groupings when names collide
I think this is really and edge case because in order to get this behavior you need to group by an external Series with the same name as one of the columns.

```
df = pd.DataFrame({"a": [1, 1, 2], "b": [3, 4, 5]})

print(df.groupby(pd.Series([6, 6, 7], name="a"), as_index=False).sum())
#    a  b
# 0  2  7
# 1  2  5
print(df.groupby(pd.Series([6, 6, 7], name="a"), as_index=True).sum().reset_index())
# ValueError: cannot insert a, already exists
```

I think users expect `as_index=False` to behave the same as calling reset_index() after using `as_index=True`. It is not good behavior that we silently drop the groupings when `as_index=False`.

### PR DESCRIPTION
**Pull request by sunwooma**
- [x] closes #58337
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/arrays/datetimes.py [hunk 1]
```diff
@@ -1628,9 +1628,8 @@ def isocalendar(self) -> DataFrame:
 
         See Also
         --------
-        DatetimeIndex.day: The day of the datetime.
-        DatetimeIndex.minute: The minutes of the datetime.
-        DatetimeIndex.second: The seconds of the datetime.
+        Series.dt.minute : Minutes of the datetime.
+        Series.dt.second : Seconds of the datetime.
```
Comment (Aloqeely) [inline pandas/core/arrays/datetimes.py:1632]: Could you split this PR into 2 separate parts?
I.E. let this PR fix the bug, and make a new PR for improving the docstrings

File: pandas/core/frame.py [hunk 2]
```diff
@@ -21,6 +21,7 @@
     Sequence,
 )
 import functools
+import pandas as pd
```
Comment (Aloqeely) [inline pandas/core/frame.py:24]: You can remove this import, line 181 imports `Series` already

File: pandas/core/arrays/datetimes.py (hunk 1 above)
Comment (sunwooma) [reply pandas/core/arrays/datetimes.py:1632]: Ok, will do.

File: pandas/core/frame.py (hunk 2 above)
Comment (sunwooma) [reply pandas/core/frame.py:24]: Will do.

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.

**Pull request closed without merging**